### PR TITLE
[10.0][MIG] stock_product_location_sorted_by_qty

### PR DESCRIPTION
**Pull request by chafique-delli**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Travis is still red

**Comment by chafique-delli** [comment]
@pedrobaeza , Hi, looking at the log file, the errors does not affect my module but other modules of the branch. You know why ?

**Comment by pedrobaeza** [comment]
Oh, yes, this was a change in the linter for being stricter on READMEs. Maybe you can fix it in another PR? :angel:

**Comment by bealdav** [comment]
Hi @chafique-delli could you participate to fix warnings in this repo ?
Consider as if you contribute as oca sprint ;-)

Pedro is ready to merge quickly your work on this topic. Isn't it @pedrobaeza ?

Thanks

**Comment by pedrobaeza** [comment]
Ok by my side

File: stock_product_location_sorted_by_qty/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# © 2017 Chafique DELLI @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Stock Product Location Sorted by Quantity',
+    'summary': 'In the update wizard of quantities for a product, '
+               'sort the stock location by quantity',
+    'version': '10.0.1.0.0',
+    'category': 'Warehouse Management',
+    'website': 'http://akretion.com',
```
Comment (MeetKD) [inline stock_product_location_sorted_by_qty/__manifest__.py]: Can you please change the  website key ?

**Review by MeetKD — CHANGES_REQUESTED** [review]

File: stock_product_location_sorted_by_qty/models/stock.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    _inherit = 'stock.location'
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        res = super(StockLocation, self).name_search(
+            name=name, args=args, operator=operator, limit=limit)
+        ordered_loc_list = []
+        if self.env.context.get('stock_change_product_quantity'):
+            domain = [
+                ('qty', '>', 0),
```
Comment (rousseldenis) [inline stock_product_location_sorted_by_qty/models/stock.py]: @chafique-delli I think this is not correct as the sum of quants equals the real location quantity (everybody does not install stock_no_negative module) as you can have stock == 0 with two quants (e.g.: -2 and 2).

**Review by rousseldenis — COMMENTED** [review]
Code review

File: stock_product_location_sorted_by_qty/models/stock.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            new_res = []
+            ordered_name_loc_list = []
+            domain = [
+                ('qty', '>', 0),
+                ('product_id', '=', self.env.context.get(
+                    'default_product_id', False)),
+                ('location_id', 'in', self.ids)
+            ]
+            quants = self.env['stock.quant'].with_context(
+                stock_change_product_quantity=False).read_group(
```
Comment (rousseldenis) [inline stock_product_location_sorted_by_qty/models/stock.py:51]: @chafique-delli Why changing context here ?

**Comment by rousseldenis** [comment]
@chafique-delli Tell us when ready to review

**Comment by chafique-delli** [comment]
@rousseldenis , I made the necessary fix  after your remark about quants.

**Review by rousseldenis — COMMENTED** [review]
Code review. Seems ok.
FYI, there is a new convention for module manifest https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py

And there is also a README generator. You can find readme directory template here : https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

File: stock_product_location_sorted_by_qty/models/stock.py (hunk 3 above)
Comment (chafique-delli) [reply stock_product_location_sorted_by_qty/models/stock.py:51]: @rousseldenis , To avoid recursion

**Comment by rousseldenis** [comment]
@chafique-delli maybe a little rebase to see if runbot goes green?

**Comment by chafique-delli** [comment]
@rousseldenis , in log file of runbot, we have the following warning:
" WARNING openerp_test odoo.addons.base.module.module: module stock_product_location_sorted_by_qty: description is empty ! "

**Comment by bealdav** [comment]
Ping

**Review by bealdav — APPROVED** [review]

**Review by hparfr — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]

**Comment by rousseldenis** [comment]
@MeetKD Could you update your review?

**Comment by bealdav** [comment]
It seems @MeetKD is not available. Merge ?

**Comment by rousseldenis** [comment]
As remarkes have been attended, merging

**Pull request merged**